### PR TITLE
[Studio] feat: add Topic send payload preflight

### PR DESCRIPTION
**Pull request by coder999o**
## What is the purpose of the change

Add a local payload preflight panel to the Topic send message modal. The preflight summarizes the final message body, Tag, Key, and custom properties before sending, detects malformed or duplicate properties, warns about reserved RocketMQ property names, and reuses the normalized payload for the actual send request.

## Brief changelog

- Add a message payload preview utility for body format detection, byte sizing, property parsing, reserved-key checks, and blocking issue collection.
- Show a live send preflight summary in the Topic send message modal.
- Prevent duplicate or malformed custom properties from silently reaching the send API.
- Add utility and page-level tests for payload preview, normalized send parameters, and duplicate-property blocking.

## Verifying this change

- `npm run test -- messagePayloadPreview TopicPage`
- `npm run lint`
- `npm run build`
- `git diff --check`
- `git ai status`

### CODE REVIEW

File: web/src/utils/messagePayloadPreview.ts [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+}
+
+export interface MessagePayloadPreviewInput {
+  topic?: string;
+  tag?: string;
+  key?: string;
+  body?: string;
+  propsMode: MessagePropertyMode;
+  propsText?: string;
+  properties?: MessagePropertyInput[];
```
Comment (RockteMQ-AI) [inline web/src/utils/messagePayloadPreview.ts:62]: **[i18n]** The `issue()` helper uses hardcoded Chinese strings for `title` and `description` fields (e.g., `'保留属性名'`, `'自定义属性名不能与 RocketMQ 内部属性同名'`). These should use i18n translation keys via `t()`, similar to how PR #3063 handles parser errors. Consider adding translation keys like `payloadPreview.reservedPropertyTitle` / `payloadPreview.reservedPropertyDesc` to `translations.ts`.

File: web/src/utils/messagePayloadPreview.ts [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    const parsed = JSON.parse(trimmed) as unknown;
+    if (Array.isArray(parsed)) return 'json-array';
+    if (parsed !== null && typeof parsed === 'object') return 'json-object';
+    return 'json-scalar';
+  } catch {
+    return 'plain-text';
+  }
+};
+
+const fromEntries = (entries: MessagePropertyPreviewEntry[]): Record<string, string> =>
```
Comment (RockteMQ-AI) [inline web/src/utils/messagePayloadPreview.ts:155]: **[Robustness]** The `fromEntries` function correctly handles prototype pollution (tested in `buildMessagePropertiesFromRows`). The implementation is safe because `Object.fromEntries` creates own properties. Consider adding a brief comment explaining why this is safe, to help future maintainers.

**Review by RockteMQ-AI — COMMENTED** [review]
### Summary
This PR adds a comprehensive message payload preflight panel to the Topic send message modal. It includes body format detection, byte sizing, property parsing, reserved-key checks, and duplicate property detection. The utility is well-structured with good test coverage.

### Findings
- **[Warning]** Hardcoded Chinese strings in `messagePayloadPreview.ts` — the `issue()` helper should use i18n translation keys instead of hardcoded strings for `title` and `description` fields.
- **[Info]** Good separation of concerns: utility handles validation logic, UI handles presentation.
- **[Info]** Comprehensive test coverage including edge cases (prototype pollution, sparse arrays, duplicate keys).
- **[Info]** Reserved property name detection is a valuable safety feature.

### Suggestions
1. Replace hardcoded Chinese strings in `messagePayloadPreview.ts` with i18n translation keys
2. Consider adding a brief comment in `fromEntries` explaining why it's safe from prototype pollution

---
*Automated review by github-manager-bot*
